[Cliff Koes] iP

Domokunx opened the pull request:
# Bibi
Easy to use task manager to help you keep your life on track!
- Simple
- Friendly
- [All at the discounted price of $0!](https://youtu.be/dQw4w9WgXcQ?si=x3zVRgRzT6JCerIk) 😎 

## Commands
1. `todo <description>`: Add tasks for the bot to keep track of 
2. `list`: Check your current list of tasks 
3. `mark <index>`: Mark/Unmark them as completed 
4. Bottom Text!

## Features (In Progress)
- [x] : ToDo, Deadline, Event task subtypes
- [ ] : ~~Sort list by deadline~~
- [ ] : Keep different tasklists

```
                          ########   #######   ########   #######
                          #       #     #      #       #     #   
                          ########      #      ########      #   
                          #       #     #      #       #     #   
                          #       #     #      #       #     #   
                          ########   #######   ########   #######
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Domokunx committed:
Renamed chatbot

Domokunx committed:
Added farewell text

Domokunx committed:
Added echo of input

Domokunx committed:
Added add and list feature

Domokunx committed:
Added Task class

Domokunx committed:
Added marking feature

Domokunx committed:
Added ToDo task tracker

Domokunx committed:
Added Deadline task tracker

Domokunx committed:
Added Event task tracker

Domokunx committed:
Added testing for current features

Domokunx committed:
Added invalid syntax warning for ToDo, Deadline and Event

Domokunx committed:
Stricter Regex Checker
Code Cleanip

Domokunx committed:
Switched from fixed array to arraylist to store Tasks

Domokunx committed:
Code Cleanup

Fixed GUI Horizontal line bugs

Domokunx committed:
Added remove cmd

Domokunx committed:
Refactored code to reduce amount of unnecessary function calls

Domokunx committed:
Fixed indentation issue

Domokunx committed:
Fix switch-case indentation

Domokunx committed:
Add initialize data folder feature

Domokunx committed:
Add read from list.txt feature

Domokunx committed:
Add save/write to file feature

Domokunx committed:
Merge branch 'branch-level-7'

Domokunx committed:
Update Deadline tasks to parse and format yyyy-mm-dd

Domokunx committed:
Merge branch 'branch-Level-8'

Domokunx committed:
Add TaskList class

Domokunx committed:
Add Storage class

Domokunx committed:
Refactor Bibi.java to run from one instance

Domokunx committed:
Refactor code for Bibi.java, Storage.java, TaskList.java, Ui.java

Domokunx committed:
Add Command.java, Parser.java classes

Domokunx committed:
Redesign code to adhere to OOP

Domokunx committed:
Refactor classes into separate packages

Domokunx committed:
Fix import statement for classes under bibi.task package

Domokunx committed:
Merge branch 'branch-Packages'

Domokunx committed:
Merge branch 'add-gradle-support'

Domokunx committed:
Fix todo command setting 	odo as description

Domokunx committed:
Sync todo command fix

Domokunx committed:
Update build.gradle to run Bibi

Domokunx committed:
Update build.gradle

Domokunx committed:
Merge branch 'add-gradle-support'

Domokunx committed:
Add JUnit tests for TaskList class

Domokunx committed:
Add JUnit test for Parser class

Domokunx committed:
Merge branch 'branch-A-JUnit'

Domokunx committed:
Update jar archiveFileName

Domokunx committed:
Untrack build.gradle file

Domokunx committed:
Update .gitignore

Domokunx committed:
Add header comments to Bibi.java methods

Domokunx committed:
Add header comments to Ui.java

Domokunx committed:
Update Ui.java to follow code style

Domokunx committed:
Add find command

Domokunx committed:
Merge branch 'branch-A-JavaDoc'

Domokunx committed:
Merge branch 'branch-A-CodingStandard'

Domokunx committed:
Merge branch 'branch-Level-9'

Eru111 commented on src/main/java/bibi/Command.java:
```diff
@@ -0,0 +1,189 @@
+
+public class Command {
+    private String cmd;
+    private String args;
```
Perhaps a more intuitive variable names here (like command and arguments)?

Eru111 commented on src/main/java/bibi/task/Deadline.java:
```diff
@@ -0,0 +1,27 @@
+                    date.getDayOfMonth(), date.getMonth().name(), date.getYear());
+        } else {
+            this.deadline = deadline;
+        };
```
is the semicolon after the curly brace here necessary?

Eru111 commented on src/main/java/bibi/Ui.java:
```diff
@@ -0,0 +1,140 @@
+import bibi.task.Task;
+import bibi.task.TaskList;
+import java.util.Scanner;
+
```
Should different groups of imports be separated by a whitespace and ordered differently? I noticed this issue in several other places.
e.g.
```
import java.util.Scanner;

import bibi.task.Task;
import bibi.task.TaskList;
```

Eru111 commented on src/main/java/bibi/Storage.java:
```diff
@@ -0,0 +1,91 @@
+package bibi;
+
+import bibi.task.*;
```
Should this wildcard import be removed as it does not follow the coding standard?

Eru111 commented on src/test/java/bibi/ParserTest.java:
```diff
@@ -0,0 +1,54 @@
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Should this static import be placed before the other import statements?

Eru111 reviewed (commented):
I like how clean the code is overall.

Domokunx committed:
Add automated checkstyle

Domokunx committed:
Merge branch 'branch-A-CheckStyle'

Domokunx committed:
Add javaFx GUI

Domokunx committed:
Merge branch 'branch-Level-10'

Domokunx committed:
Modify find command to use varargs

Domokunx committed:
Fix GUI command responses

Domokunx committed:
Merge branch 'branch-A-Varargs'

Xczheng0105 commented on src/main/java/bibi/Command.java:
```diff
@@ -0,0 +1,115 @@
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
+        int index;
+        // Preconfigured commands
+        switch (cmd) {
```
Correct style for switch-case indentation. Good =)

Xczheng0105 commented on src/main/java/bibi/Storage.java:
```diff
@@ -0,0 +1,127 @@
+package bibi;
+
+import java.io.File;
```
Imports are organised, no * imports used. Good =)

Xczheng0105 commented on src/main/java/bibi/Storage.java:
```diff
@@ -0,0 +1,127 @@
+                }
+
+                break;
+            default:
```
Good implementation of a default case here.

Xczheng0105 commented on src/main/java/bibi/Ui.java:
```diff
@@ -0,0 +1,295 @@
+package bibi;
+
+import java.io.IOException;
```
Try to separate your import statements into packages that are related, and try to order them alphabetically.

Xczheng0105 commented on src/main/java/bibi/gui/Main.java:
```diff
@@ -0,0 +1,66 @@
+package bibi.gui;
+
+import java.io.IOException;
```
Once again, do watch the alphabetical ordering.

Xczheng0105 commented on src/main/java/bibi/gui/MainWindow.java:
```diff
@@ -0,0 +1,53 @@
+package bibi.gui;
+
+import bibi.Bibi;
```
Should space out the import bibi.Bibi line from the other import statements.

Xczheng0105 reviewed (commented):
Conforms to coding standard generally. LGTM =)

Domokunx committed:
Fix indentations

Currently fails checkstyle check.

Fix indentation for lines warned by checkstyle accordingly.

Maintaining a standard indentation style improves readability for other programmers.

Domokunx committed:
Refactor UI class into Processor class

Ui class is no longer used due to switch from console outputs to using a GUI.

This results in a collection of dead functions.

Removing the unused functions and and renaming Ui class to Processor makes the code more readable for other programmers to modify in the future.

Remove the unused functions and renamed Ui to Processor. Move all input processing logic from Command into Processor.

This allows for all the relevant code to be grouped together, allowing for easier modification in the future.

Domokunx committed:
Add assertion to Processor

Processor assumes all commands are valid without checking.

Add an assertation to default branch in case an invalid command passes through all the checks.

Allows for us to find the bug earlier before things get problematic.

Domokunx committed:
Remove unused functions in Task class

Domokunx committed:
Merge pull request #1 from Domokunx/branch-A-Assertions

Branch a assertions

Domokunx committed:
Merge branch 'master' into branch-A-CodeQuality

Domokunx committed:
Merge pull request #2 from Domokunx/branch-A-CodeQuality

Remove unused functions in Task class

Domokunx committed:
Add .yml file for GitHub Actions workflow

Domokunx committed:
Add tag C-BetterSearch

find command already supports BetterSearch.

Domokunx committed:
Fix build issue

Domokunx committed:
Move data folder to root

Domokunx committed:
Add Circle Clip to profile image

Domokunx committed:
Add resizable nodes and minimum main window size

Domokunx committed:
Add error display for clearer visual for user

Domokunx committed:
Add DialogBox ability to resize based on text content

Domokunx committed:
Add new profile pictures

Domokunx committed:
Merge branch 'branch-A-BetterGui'

Domokunx committed:
Update README.md

Domokunx committed:
Add Ui.png to docs folder

Domokunx committed:
Merge branch 'master' of https://github.com/Domokunx/ip

Domokunx committed:
Fix help command indentation issue

Domokunx committed:
Merge branch 'branch-A-BetterGui'

Domokunx committed:
Update README.md

Domokunx committed:
Merge branch 'branch-A-UserGuide'

Domokunx committed:
Add todo, remove commands to User Guide

Domokunx committed:
Add deadline, event commands to User Guide

Domokunx committed:
Add find command to User Guide

Domokunx committed:
Merge branch 'branch-A-UserGuide'

Domokunx committed:
Fix tip section overlapping with help command subheading

Domokunx committed:
Make example image larger for easier viewing

Domokunx committed:
Fix tip section overlapping with find syntax

Domokunx committed:
Fix CI workflow unable to detect build.gradle file

Domokunx committed:
Add Parser JUnit test

Domokunx committed:
Reconfigure to SDK 17

Domokunx committed:
Update gradle properties

Domokunx committed:
Add BibiRuntimeExceptions and handle them

Domokunx committed:
Merge branch 'branch-A-MoreErrorHandling'

Domokunx committed:
Fix checkstyle tests
